src/model-cache.ts

```diff
@@ -109,30 +109,29 @@ export default class ModelCache implements IModelCache {
         const results = await generator();
 
         function getData(data: any) {
-            if (results === null) {
+            if (data === null) {
                 return 'null'; // JSON stringified "null"
             }
 
             if (options.raw || options.plain) {
                 return JSON.stringify(data);
             }
 
-            if (!results) {
-                return results;
+            if (typeof data === 'number') {
```

Comment:
I have simply changed the order here where i do the number check before the `!data` check.. if someone where to reverse it again, a test will fail.